server/plugin/webhook.go

```diff
@@ -657,6 +657,7 @@ func (p *Plugin) postPushEvent(event *github.PushEvent) {
 		return
 	}
 
+	setShowAuthorInCommitNotification(p.configuration.ShowAuthorInCommitNotification)
```

Comment:
This seems a bit weird to me that this is used as a package-level var that's set in this function to be implicitly used by the template functions. Is there a more graceful way to "inject" this config value into the logic? I suppose it's not a big deal since the config value will mostly stay the same, so no concerns about race conditions etc., and it is "reset" here for every webhook request. @hanzei What are your thoughts on this?

Reply:
Yes, I spent a while thinking about this. 

Take a look at the Golang text/template API: https://pkg.go.dev/text/template I don't see a way to inject "context" other than the main data object itself. This would require defining a new data structure which includes the github.Event *and* some other attributes like "ShowAuthorInCommitNotification".

Also, while looking at template.go in general, there is already an established way of injecting context, the `gitHubToUsernameMappingCallback`, which is a package-level var: https://github.com/mattermost/mattermost-plugin-github/blob/2608197665120cdd99d5deae9fb01fe7bee6f20e/server/plugin/template.go#L34

So, I decided to follow the existing design pattern in the file, and requires the least invasive changes to code not directly related to my change.

Reply:
Sounds good to me :+1: Thanks @cpatulea!